docs/proposals/20191016-clusterctl-redesign.md

```diff
@@ -0,0 +1,271 @@
+
+As of this writing, we envision clusterctl’s component model to look similar to the diagram below:
+
+![components](images/clusterctl-redesign/components.png)
+
+Clusterctl consists of a single binary artifact implementing a CLI; the same features exposed by the CLI will be made available to other management tools via a client library.
+
+During provider installation, clusterctl internal pkg/lib is expected to access provider repositories and read the yaml file specifying all the provider components to be installed in the management cluster.
+
+If the provider repository contains deployment templates, or flavors, clusterctl could be used to generate yaml for creating new workload clusters.  
+
+Behind the scenes clusterctl will apply labels to any and all providers’ components and use a set of custom objects for keeping track of some metadata about installed providers, like e.g. the current version or the namespace where the components are installed. 
+
+#### Init sequence
+
+The clusterctl CLI is optimized for the “day one” experience of using Cluster API, and it will be possible to create a first cluster with two commands after the pre-requisites are met:
+
+```shell
+$ clusterctl init --infrastructure aws
+$ clusterctl config cluster my-first-cluster | kubectl apply -f -
+```
+
+Then, as of today, the CNI of choice should be applied to the new cluster:
+
+```shell
+$ kubectl get secret my-first-cluster-kubeconfig -o=jsonpath='{.data.value}' | base64 -D > my-first-cluster.kubeconfig
+$ kubectl apply --kubeconfig=my-first-cluster.kubeconfig -f MY_CNI
+```
+
+The internal flow for the above commands is the following:
+
+**clusterctl init (AWS):**
+
+![components](images/clusterctl-redesign/init.png)
+
+Please note that “day one” experience:
+
+1. Is based on a list of pre-configured provider repositories; it is possible to add new provider configurations to the list of known providers by using the clusterctl config file (see day 2 operations for more details about this).
+2. Assumes init will replace env variables defined in the components yaml read from provider repositories, or error out if such variables are missing.
+3. Assumes providers will be installed in the namespace defined in the components yaml; at the time of this writing, all the majoir providers ships with components yaml creating dedicated namespaces. This could be customized by specifying the --target-namespace flag.
+4. Assumes to not change the namespace each provider is watching on objects; at the time of this writing, all the majoir providers ships with components yaml watching for objects in all namespaces. Pending [1490](https://github.com/kubernetes-sigs/cluster-api/issues/1490) issue addressed, this could be customized by specifying the --watching-namespace flag.
```

Comment:
```suggestion
4. Assumes to not change the namespace each provider is watching on objects; at the time of this writing, all the major providers ship with components yaml watching for objects in all namespaces. Pending [1490](https://github.com/kubernetes-sigs/cluster-api/issues/1490) issue addressed, this could be customized by specifying the --watching-namespace flag.
```